File: 0162-find-peak-element/0162-find-peak-element.ts
function findPeakElement(nums: number[]): number {
    let n = nums.length;

    for(let i = 0; i < n; i++) {
        if((i === 0 || nums[i] > nums[i-1]) && (i === n - 1 || nums[i] > nums[i + 1])) {
            return i;
        }
    }

    return n - 1;
};

File: 0162-find-peak-element/0162-find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& arr) {
        int n = arr.size();

        if(n == 1 || arr[0] > arr[1]) {
            return 0;
        }

        if(arr[n-1] > arr[n-2]) {
            return n - 1;
        }

        int low = 1;
        int high = n - 2;
        int mid;

        while(low <= high) {
            mid = (low + high)/2;

            if(arr[mid] > arr[mid - 1] && arr[mid] > arr[mid + 1]) {
                return mid;
            } else if(arr[mid - 1] <= arr[mid]) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        return mid;
    }
};